x/concentrated-liquidity/keeper_test.go

```diff
@@ -313,3 +310,25 @@ func (s *KeeperTestSuite) validateListenerCallCount(
 func (s *KeeperTestSuite) setListenerMockOnConcentratedLiquidityKeeper() {
 	s.App.ConcentratedLiquidityKeeper.SetListenersUnsafe(types.NewConcentratedLiquidityListeners(&clmocks.ConcentratedLiquidityListenerMock{}))
 }
+
+// Crosses the tick and charges the fee on the global fee accumulator.
+// This mimics crossing an initialized tick during a swap and charging the fee on swap completion.
+func (s *KeeperTestSuite) crossTickAndChargeFee(poolId uint64, tickIndexToCross int64) {
+	// Cross the tick to update it.
+	_, err := s.App.ConcentratedLiquidityKeeper.CrossTick(s.Ctx, poolId, tickIndexToCross, DefaultFeeAccumCoins[0])
+	s.Require().NoError(err)
+	err = s.App.ConcentratedLiquidityKeeper.ChargeFee(s.Ctx, poolId, DefaultFeeAccumCoins[0])
+	s.Require().NoError(err)
+}
```

Comment:
For my learning, does the order of these matter? In my mind, it feels like one should charge the fee before crossing the tick, but it could be the same.

Reply:
The order matters because if we charged fee first, the tick's accumulators would be updated with the new value.

The order is consistent with our [swap logic](https://github.com/osmosis-labs/osmosis/blob/72762cc3c239dad8a1b5d64000cd0cf6111cecd9/x/concentrated-liquidity/swaps.go#L318-L393)

Note that we cross ticks as we swap, and then "charge fee" after the loop completes

Reply:
Makes sense, resolving!